board/safety/safety_mazda.h

```diff
@@ -33,29 +36,62 @@ static int mazda_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
   int bus = GET_BUS(to_push);
   int addr = GET_ADDR(to_push);
 
-  if ((addr == MAZDA_STEER_TORQUE) && (bus == MAZDA_MAIN)) {
-    int torque_driver_new = GET_BYTE(to_push, 0) - 127;
-    // update array of samples
-    update_sample(&mazda_torque_driver, torque_driver_new);
-  }
+  if (bus == MAZDA_MAIN) {
 
-  // enter controls on rising edge of ACC, exit controls on ACC off
-  if ((addr == MAZDA_CRZ_CTRL) && (bus == MAZDA_MAIN)) {
-    int cruise_engaged = GET_BYTE(to_push, 0) & 8;
-    if (cruise_engaged != 0) {
-      if (!mazda_cruise_engaged_last) {
-        controls_allowed = 1;
+    // sample speed
+    if (addr == MAZDA_WHEEL_SPEED) {
+      int speed = 0;
+      // sum 4 wheel speeds
+      for (int i=0; i<8; i+=2) {
+        int next_byte = i + 1;  // hack to deal with misra 10.8
+        speed += (GET_BYTE(to_push, i) << 8) + GET_BYTE(to_push, next_byte) - 10000;
       }
+      vehicle_moving = ABS(speed / 4) > 100;
     }
-    else {
-      controls_allowed = 0;
+
+    
+    if (addr == MAZDA_STEER_TORQUE) {
+      int torque_driver_new = GET_BYTE(to_push, 0) - 127;
+      // update array of samples
+      update_sample(&mazda_torque_driver, torque_driver_new);
     }
-    mazda_cruise_engaged_last = cruise_engaged;
-  }
 
-  // if we see wheel speed msgs on MAZDA_CAM bus then relay is closed
-  if ((safety_mode_cnt > RELAY_TRNS_TIMEOUT) && (bus == MAZDA_CAM) && (addr == MAZDA_WHEEL_SPEED)) {
-    relay_malfunction_set();
+    // enter controls on rising edge of ACC, exit controls on ACC off
+    if (addr == MAZDA_CRZ_CTRL) {
+      int cruise_engaged = GET_BYTE(to_push, 0) & 8;
+      if (cruise_engaged != 0) {
+	if (!cruise_engaged_prev) {
+	  controls_allowed = 1;
+	}
```

Comment:
indent, and fix the brackets to match the style of the rest of the panda code

Reply:
More tabs!